### ISSUE TITLE
Many components' `css` type is broken in `8.8.0-alpha.1`

### ISSUE THREAD

**Issue by tvanantwerp** [stripe/stripe-apps#862]
**Describe the bug**
Using the `8.8.0-alpha.1` version of `@stripe/ui-extension-sdk` along with TypeScript version `5.1.3`, the `css` types for components like `Link` and `Button` no long work as expected. Instead of _allowing_ for `width`, `alignX`, or both, the types now demand _both_ `width` and `alignX`.

I believe the issue is in `manual_components.d.ts`. Components like `Box` have their extra CSS properties (mainly text overflow and wrap related) explicitly written out as optional. E.g., `css?: { textOverflow?: "ellipsis" | undefined };`. However, many of the smaller components use a helper type named `ExposeCSS` that relies on the `Pick` utility type to choose CSS properties from a predefined set. `Button` example: `readonly props?: ExposeCSS<ButtonProps, "width" | "alignX"> | undefined;`

The helper `ExposeCSS` lacks the `Partial` utility type wrapping the `Pick` to allow all of those CSS properties to be option, matching the behavior of components like `Box`.

**To Reproduce**
1. Have a Stripe app with the following dependencies:
   ```
   "@stripe/ui-extension-sdk": "^8.8.0-alpha.1",
   "typescript": "5.1.3"
   ```
2. Create a component like `Link` or `Button`.
3. Pass `css` props to the component. TypeScript will complain if you do not use every available property. In the case of `Button` or `Link`, it will expect both `width` and `alignX`.

**Expected behavior**
Should be able to use a subset of allowed CSS properties on components types with the `ExposeCSS` helper type.

**Screenshots**
<img width="713" alt="image" src="https://github.com/stripe/stripe-apps/assets/2487968/187f6397-a903-42d6-9763-7464cc5c9b8e">

**Solution**

Modifying `ExposeCSS` as follows fixes the type errors:

```ts
declare type ExposeCSS<P, T extends keyof TokenizedCSSProperties> = P & {
    css?: Partial<Pick<TokenizedCSSProperties, T>>;
};
```

**Comment by ericfrank-stripe** [comment]
Hi @tvanantwerp, so that we can better diagnose the issue, what version of the SDK did you upgrade from? Regarding Tyepscript, our app generator is currently on `^4.7.4` and we haven't tried an upgrade yet, so that may be causing the issue as well.

**Comment by ericfrank-stripe** [comment]
@tvanantwerp I set up my environment with SDK `8.8.0-alpha.1` and TypeScript `5.1.3`, and I wasn't able to reproduce the errors you're seeing. Please try the following:

Download this bundle with a sample project:

[App.zip](https://github.com/stripe/stripe-apps/files/11887970/App.zip)

Unzip the project, and on your command line, `cd` into the directory and run: `yarn install`, then `yarn types`. This will run the TypeScript typechecker, which is more reliable than the IDE typecheckers in VSCode. Let me know if you get any errors in your output.

When I ran this command, I got:
```
% yarn types    ericfrank/UIXT-319
yarn run v1.22.19
$ tsc --noEmit
✨  Done in 1.08s.
```

To make sure it was working, I changed the button width to `fillz`, and running the command got me this:

```
% yarn types    ericfrank/UIXT-319
yarn run v1.22.19
$ tsc --noEmit
src/views/App.tsx:6:22 - error TS2820: Type '"fillz"' is not assignable to type '0 | "fill" | "auto" | 1 | 25 | 50 | 75 | 100 | 150 | 200 | 250 | 300 | 350 | 400 | 500 | 600 | "min" | "max" | "fit" | "1/2" | "1/3" | "2/3" | "1/4" | "2/4" | "3/4" | "1/5" | "2/5" | "3/5" | ... 17 more ... | undefined'. Did you mean '"fill"'?

6       <Button css={{ width: "fillz" }}>Button</Button>
                       ~~~~~
Found 1 error in src/views/App.tsx:6
```

Thanks for your help!

**Comment by tvanantwerp** [comment]
Hmm, this raise interesting new issues due to our use of Nx monorepos. Writing my experience here in case anyone else ends up in the same situation later after an upgrade.

---

The [migration guide](https://stripe.com/docs/stripe-apps/ui-extension-developer-tools#migrating-ui-extension-tools) assumes that the Stripe App lives alone at the root of its own repository. However, all of the apps my team has worked on are Nx monorepos with the Stripe App as just one app in a sub-directory (`apps/stripe-app`), sibling to their Node back ends (`apps/api`). These Nx repos have a single `package.json` and `tsconfig.base.json` at the root, shared between apps. This has been useful for sharing libraries between front and back end, including things like types and utility functions.

But this creates issues:

1) The Node back ends still require TypeScript, ESLint, etc. We can't blanket uninstall them like the migration guide suggests. But I can mostly get rid of React deps.
2) Attempts to implement the changes to Stripe App testing broke our tests quite badly. In the end, I did not follow through with migrating our testing setup.
3) The `tsconfig` instructions broke the Stripe App build process. The build fails to resolve any dependencies in the shared libraries outside of the Stripe App's directory. Example, where `@stripe-apps-template/api-interfaces` is mapped by the root `tsconfig.json` to `libs/api-interfaces`:
  ```
  ✘ [ERROR] Could not resolve "@stripe-apps-template/api-interfaces"
      src/views/Settings.tsx:4:7:
        4 │ } from '@stripe-apps-template/api-interfaces';
          ╵        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    You can mark the path "@stripe-apps-template/api-interfaces" as external to exclude it from the bundle, which will remove this error.
  ```

This error occurs because the Stripe App's `tsconfig.json` no longer extends the root level `tsconfig.json` that specifies how to resolve `@stripe-apps-template/api-interfaces`. It looks like [multiple `tsconfig.json` extends](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#supporting-multiple-configuration-files-in-extends) was added in TypeScript v5, but it doesn't seem to work when running `stripe apps start --manifest stripe-app.development.json`. Both `ui-extension-sdk` and `ui-extension-tools` are on versions of TypeScript 4, so I'm assuming the TypeScript version used when running the app via CLI is TypeScript 4 too.

For now, I can manually copy the contents of `@stripe/ui-extension-tools/tsconfig.ui-extension` to my Stripe App's `tsconfig.json`, as it's not a lot and mostly overlapped our own. But this opens the possibility of getting out of sync with Stripe's chosen configuration again in the future. And this doesn't solve the problem that I'm now likely to be running type checks with a different version of TypeScript than the one assumed by either `@stripe/ui-extension-tools` or `@stripe/ui-extension-sdk`. For example, I get this warning now when linting a Stripe App:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.1.5

Please only submit bug reports when using the officially supported version.

=============
```
